packages/delisp-core/src/convert.ts

```diff
@@ -293,6 +306,45 @@ defineToplevel("export", expr => {
   };
 });
 
+defineToplevel("type", expr => {
+  const [typeOp, ...args] = expr.elements;
+
+  if (args.length !== 2) {
+    const lastExpr = last([typeOp, ...args]) as ASExpr;
+    throw new Error(
+      printHighlightedExpr(
+        `'type' needs exactly 2 arguments, got ${args.length}`,
+        lastExpr.location,
+        true
+      )
+    );
+  }
+
+  const [name, definition] = args;
+
+  if (name.type !== "symbol") {
+    throw new Error(
+      printHighlightedExpr("'type' expected a symbol as a name", name.location)
+    );
+  }
+
+  checkUserDefinedTypeName(name);
+
+  const definitionType = convertType(definition);
+  if (listTypeVariables(definitionType).length > 0) {
+    throw new Error(
+      printHighlightedExpr("Type variable out of scope", definition.location)
+    );
+  }
+
+  return {
+    type: "type-alias",
```

Comment:
Lets maybe rename this to `type-definition`?

Reply:
And have a `alias: boolean` property?  I guess we can do that when we introduce normal type definitions? I'm not sure how different they will be at syntax level. Most likely we'll need a new keyword or variant

```
(data ID
  (ID number))
```
?